pkg/volume/csi/csi_plugin.go

```diff
@@ -324,24 +371,44 @@ func (p *csiPlugin) ConstructVolumeSpec(volumeName, mountPath string) (*volume.S
 	}
 
 	klog.V(4).Info(log("plugin.ConstructVolumeSpec extracted [%#v]", volData))
-
-	fsMode := api.PersistentVolumeFilesystem
-	pv := &api.PersistentVolume{
-		ObjectMeta: meta.ObjectMeta{
-			Name: volData[volDataKey.specVolID],
-		},
-		Spec: api.PersistentVolumeSpec{
-			PersistentVolumeSource: api.PersistentVolumeSource{
-				CSI: &api.CSIPersistentVolumeSource{
-					Driver:       volData[volDataKey.driverName],
-					VolumeHandle: volData[volDataKey.volHandle],
+	var spec *volume.Spec
+
+	if p.inlineFeatureEnabled && volData[volDataKey.sourceKind] == volumeKind {
```

Comment:
these checks seem redundant since you just set sourceKind based on the `inlineFeatureEnabled`. They essentially mean the same thing?